gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.DNS_NAME)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNotNull(zone.dnsName());
+    assertNull(zone.description());
+    assertNull(zone.nameServerSet());
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.DESCRIPTION)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNotNull(zone.description());
+    assertNull(zone.nameServerSet());
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.NAME_SERVERS)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNull(zone.description());
+    assertNull(zone.nameServerSet());
+    assertTrue(!zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    zones = ImmutableList.copyOf(DNS.listZones(Dns.ZoneListOption.dnsName(ZONE1.dnsName()),
+        Dns.ZoneListOption.fields(Dns.ZoneField.NAME_SERVER_SET)).iterateAll());
+    assertEquals(1, zones.size());
+    zone = zones.get(0);
+    assertNull(zone.creationTimeMillis());
+    assertNotNull(zone.name());
+    assertNull(zone.dnsName());
+    assertNull(zone.description());
+    assertNull(zone.nameServerSet()); // we cannot set it using gcloud java
+    assertTrue(zone.nameServers().isEmpty());
+    assertNull(zone.id());
+    // several combined
+    zones = filter(DNS.listZones(Dns.ZoneListOption.fields(Dns.ZoneField.ZONE_ID,
+        Dns.ZoneField.DESCRIPTION),
+        Dns.ZoneListOption.pageSize(1)).iterateAll());
+    assertEquals(2, zones.size());
+    for (Zone current : zones) {
+      assertNull(current.creationTimeMillis());
+      assertNotNull(current.name());
+      assertNull(current.dnsName());
+      assertNotNull(current.description());
+      assertNull(current.nameServerSet());
+      assertTrue(zone.nameServers().isEmpty());
+      assertNotNull(current.id());
+    }
+  }
+
+  @Test
+  public void testDeleteZoneUsingServiceObject() {
+    Zone created = DNS.create(ZONE1);
+    assertEquals(created, DNS.getZone(ZONE1.name()));
+    DNS.delete(ZONE1.name());
+    assertNull(DNS.getZone(ZONE1.name()));
+  }
+
+  @Test
+  public void testDeleteZoneUsingZoneObject() {
```

Comment:
Nit: also here I am not sure we need this given the previous method and unit tests.